canonical-web-and-design/maas-ui
# Add table checkbox helper functions and array utils.

## Done

- Added `generateCheckboxHandlers` helper function that generates `handleRowCheckbox` and `handleGroupCheckbox` functions. `handleAllCheckbox` has been removed and is simply a form of `handleGroupCheckbox` where the group is all of the machines/kvms/etc
- Added array utils `someInArray` and `someNotAll` which are fully abstracted forms of the old `checkboxChecked` and `checkboxMixed` functions.
- Changed the `machine.setSelected` action so that you only need the machine system_ids for it to work.

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Go to /MAAS/r/machines and check that the machine selection still works correctly. Test that selecting individual machines, groups and the header checkbox all work. Also check that the "X of Y machines cannot perform action" still filters the list correctly, and that deselecting everything removes the selected filter.
- Do the same on /MAAS/r/kvm.
